dataland-frontend/src/utils/DataTableDisplay.ts

```diff
@@ -30,16 +30,90 @@ export function compareReportingPeriods(firstReportingPeriod: string, secondRepo
     return -1;
   } else if (!isNaN(Number(secondReportingPeriod))) {
     return 1;
+  } else if (firstReportingPeriod > secondReportingPeriod) {
+    return 1;
   } else {
-    if (firstReportingPeriod > secondReportingPeriod) {
-      return 1;
-    } else {
-      return -1;
-    }
+    return -1;
   }
 }
 
 export type ReportingPeriodOfDataSetWithId = {
   dataId: string;
   reportingPeriod: string;
 };
+
+/**
+ * Adds click event listeners on DataTable row headers to expand and collapse row
+ * @param expandedRowsOnClick function passes the latest list of expanded row id's
+ * @param newExpandedRowsCallback function that returns the updated (after click) list of expanded row id's
+ * @returns the map of rows and their click handlers needed for unmounting
+ */
+export function mountRowHeaderClickEventListeners(
+  expandedRowsOnClick: () => string[],
+  newExpandedRowsCallback: (newExpandedRows: string[]) => void,
+): Map<Element, EventListener> {
+  const handlerMap: Map<Element, EventListener> = new Map();
+  let expandedRowGroups: string[] = [];
+
+  const buttonAttributeName = "data-parent-id";
+  const rowHeaders = Array.from(document.querySelectorAll("[data-row-header-click]"));
+  const rowButtons = rowHeaders
+    .map((el: Element) => {
+      const button = el.parentNode?.querySelector('button[data-pc-section="rowgrouptoggler"]');
+      if (button) {
+        button.setAttribute(buttonAttributeName, el.id);
+        return button;
+      }
+      return void 0;
+    })
+    .filter((button): button is HTMLButtonElement => !!button);
+
+  [...rowHeaders, ...rowButtons].forEach((el: Element) => {
+    let clickHandler: EventListener | null = (evt): void => {
+      if (!el.id) {
+        evt.stopImmediatePropagation();
+      }
+
+      expandedRowGroups = expandedRowsOnClick();
+      if (!expandedRowGroups.includes(el.id)) {
+        expandedRowGroups.push(el.id);
+      } else {
+        expandedRowGroups = expandedRowGroups.filter((id: string) => id !== el.id);
+      }
+      newExpandedRowsCallback(expandedRowGroups);
+    };
+
+    let target;
+    if (el?.getAttribute(buttonAttributeName)) {
+      target = el;
+    } else {
+      target = el.parentNode;
+    }
+
+    if (target) {
+      handlerMap.set(target as Element, clickHandler);
+      target.addEventListener("click", clickHandler);
+    }
+
+    clickHandler = null;
```

Comment:
Why is this setting of null needed?   This is a forEach loop and on each loop, you re-assign "clickHandler" again on line 72.  I don't think that you need to set it to null at the end of each loop.

If I am right, you could also make "clickHandler" a `const`.

Reply:
But the clickHandler variable is inside of a block and therefore a block-scoped variable, meaning that JavaScript should delete it from memory once the block is left (because the variable can't be reached anymore).

Therefore I suggest removing it.

Reply:
Done.